packages/react/src/hooks/actions/useStateMutationAction.ts

```diff
@@ -0,0 +1,37 @@
+import * as React from 'react';
+import { Hub } from '@aws-amplify/core';
+
+import {
+  ACTIONS_CHANNEL,
+  ACTION_STATE_MUTATION_FINISHED,
+  ACTION_STATE_MUTATION_STARTED,
+} from './constants';
+
+type UseStateMutationAction<T> = [T, (newState: T) => void];
```

Comment:
I'm not a big fan of single letter Type variables. Please give T a name, like `StateType`.

Reply:
It seems we normally would use a single letter for generic type, which I do see the value by differentiating it from other interfaces or types when navigating through the code

Reply:
It's standard for generics in the Typescript guide, but I think it's bad practice because it doesn't say anything about what Type is supposed to be passed in. The `<>` indicate that it's a generic, I don't think we have to be vague about what the type should be. 
See: https://twitter.com/kentcdodds/status/1083831867294441472
https://fettblog.eu/tidy-typescript-name-your-generics/

Reply:
Updated!